client/tx/tx.go

```diff
@@ -34,6 +34,13 @@ func GenerateOrBroadcastTxCLI(clientCtx client.Context, flagSet *pflag.FlagSet,
 // GenerateOrBroadcastTxWithFactory will either generate and print and unsigned transaction
 // or sign it and broadcast it returning an error upon failure.
 func GenerateOrBroadcastTxWithFactory(clientCtx client.Context, txf Factory, msgs ...sdk.Msg) error {
+	// Validate all msgs before generating or broadcasting the tx.
```

Comment:
In v0.42.x, we were calling ValidateBasic in each CLI handler:

https://github.com/cosmos/cosmos-sdk/blob/75f84b5e161bca07bc842671928713a32fa02713/x/bank/client/cli/tx.go#L51-L54

In #8512, we factorized to call ValidateBasic in `SvcMsgClientConn`, see [example](https://github.com/cosmos/cosmos-sdk/pull/8512/files#diff-49b960b21d6e2e0039e3f9a41e4c7143c5e83ec5a6365513d69b8b12e98bb336L72).

In this PR, we a re removing `SvcMsgClientConn`. We can:
- either go back to to what we did in v0.42.x, i.e. call ValidateBasic in each CLI handler.
- or factorize ValidateBasic in `GenerateOrBroadcastTxWithFactory`.

I went with the 2nd choice because it's less code, wdyt?